modules/packages/maven/metadata.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package maven
+
+import (
+	"encoding/xml"
+	"io"
+
+	"code.gitea.io/gitea/modules/validation"
+)
+
+// Metadata represents the metadata of a Maven package
+type Metadata struct {
+	GroupID      string        `json:"group_id"`
+	ArtifactID   string        `json:"artifact_id"`
+	Name         string        `json:"name"`
+	Description  string        `json:"description"`
+	ProjectURL   string        `json:"project_url"`
+	Licenses     []string      `json:"licenses"`
+	Dependencies []*Dependency `json:"dependencies"`
+}
+
+// Dependency represents a dependency of a Maven package
+type Dependency struct {
+	GroupID    string `json:"group_id"`
+	ArtifactID string `json:"artifact_id"`
+	Version    string `json:"version"`
+}
+
+type pomStruct struct {
+	XMLName     xml.Name `xml:"project"`
+	GroupID     string   `xml:"groupId"`
+	ArtifactID  string   `xml:"artifactId"`
+	Version     string   `xml:"version"`
+	Name        string   `xml:"name"`
+	Description string   `xml:"description"`
+	URL         string   `xml:"url"`
+	Licenses    []struct {
+		Name         string `xml:"name"`
+		URL          string `xml:"url"`
+		Distribution string `xml:"distribution"`
+	} `xml:"licenses>license"`
+	Dependencies []struct {
```

Comment:
I'm no Maven expert but does the client really trust the server informations about the dependencies? I would expect the client downloads the package and checks the local `pom` file for all needed informations.

Reply:
As I already said, I am also unsure whether they are needed.
Potentially yes, potentially no.
It could, however, be that the package cannot be registered in case advanced properties are used in the pom file. That's what I'm uncertain about.
In the worst case, this will simply be added to the data model in another PR, after this PR has been merged and tested thoroughly.

Reply:
The `pom` file is not generated but the original file is stored. So all metadata is only used for the Gitea UI. Currently I don't think there is a problem.